sdk/core/core-amqp/src/errors.ts

```diff
@@ -547,65 +610,74 @@ function isBrowserWebsocketError(err: any): boolean {
+  "InsufficientCreditError",
+
+  // Defines the error that occurs when the Sender fails to send a message.
+  "SendOperationFailedError"
+];
+
 /**
- * Translates the AQMP error received at the protocol layer or a generic Error into a MessagingError.
+ * Translates the AQMP error received at the protocol layer or a SystemError into a MessagingError.
+ * All other errors are returned unaltered.
  *
  * @param {AmqpError} err The amqp error that was received.
  * @returns {MessagingError} MessagingError object.
  */
-export function translate(err: AmqpError | Error): MessagingError {
-  if ((err as MessagingError).translated) {
-    // already translated
-    return err as MessagingError;
-  }
-
-  let error: MessagingError = err as MessagingError;
-
+export function translate(err: AmqpError | Error): MessagingError | Error {
   // Built-in errors like TypeError and RangeError should not be retryable as these indicate issues
   // with user input and not an issue with the Messaging process.
   if (err instanceof TypeError || err instanceof RangeError) {
-    error.retryable = false;
-    return error;
+    return err;
   }
 
   if (isAmqpError(err)) {
     // translate
-    const condition = (err as AmqpError).condition;
-    const description = (err as AmqpError).description as string;
-    error = new MessagingError(description);
+    const condition = err.condition;
+    const description = err.description!;
+    const error = new MessagingError(description);
     if ((err as any).stack) error.stack = (err as any).stack;
-    error.info = (err as AmqpError).info;
-    error.condition = condition;
+    error.info = err.info;
     if (condition) {
-      error.name = ConditionErrorNameMapper[condition as keyof typeof ConditionErrorNameMapper];
+      error.code = ConditionErrorNameMapper[condition as keyof typeof ConditionErrorNameMapper];
     }
-    if (!error.name) error.name = "MessagingError";
     if (
       description &&
       (description.includes("status-code: 404") ||
         description.match(/The messaging entity .* could not be found.*/i) !== null)
     ) {
-      error.name = "MessagingEntityNotFoundError";
+      error.code = "MessagingEntityNotFoundError";
     }
-    if (retryableErrors.indexOf(error.name) === -1) {
+    if (error.code && retryableErrors.indexOf(error.code) === -1) {
       // not found
       error.retryable = false;
     }
     return error;
   }
 
+  if (err.name === "MessagingError") {
+    // already translated
+    return err;
+  }
+
   if (isSystemError(err)) {
     // translate
     const condition = (err as any).code;
-    const description = (err as Error).message;
-    error = new MessagingError(description);
-    if ((err as any).stack) error.stack = (err as any).stack;
+    const description = err.message;
+    const error = new MessagingError(description, err);
+    if (err.stack) error.stack = err.stack;
```

Comment:
`stack` is part of `systemErrorFieldsToCopy`, so we don't need the above?

Reply:
Correct, removed.